llvm/lib/DWP/DWP.cpp

```diff
@@ -232,9 +238,12 @@ static Error addAllTypesFromDWP(
     static_assert(sizeof(OldOffset) == sizeof(TypesOffset));
     TypesOffset += C.getLength();
     if (OldOffset > TypesOffset) {
-      if (Error Err = sectionOverflowErrorOrWarning(
-              OldOffset, TypesOffset, "Types", ContinueOnCuIndexOverflow))
+      if (Error Err = sectionOverflowErrorOrWarning(OldOffset, TypesOffset,
+                                                    "Types", OverflowOptValue,
+                                                    AnySectionOverflow))
         return Err;
+      if (AnySectionOverflow)
```

Comment:
Doesn't seem like this check is necessary.  Either error will be returned that needs to be handled or return is success in which case it will go to return on line 249.

Reply:
The consideration here is to allow for an early return within the loop, so we can avoid further computations on the offset and Out.emitBytes. Also, it seems that we need to revert the offset back to its not overflowed value.

Reply:
ah ok missed loop